crates/jstzd/src/task/octez_client.rs

```diff
@@ -0,0 +1,199 @@
+use super::{endpoint::Endpoint, octez_node::DEFAULT_RPC_ENDPOINT};
+use anyhow::{bail, Result};
+use http::Uri;
+use std::{path::PathBuf, str::FromStr};
+use tempfile::{tempdir, TempDir};
+
+const DEFAULT_BINARY_PATH: &str = "octez-client";
+#[derive(Default)]
+pub struct OctezClientBuilder {
+    // if None, use the binary in $PATH
+    binary_path: Option<PathBuf>,
+    // if None, use temp directory
+    base_dir: Option<PathBuf>,
+    // if None, use localhost:8732 (the default endpoint for octez-node)
+    endpoint: Option<Endpoint>,
+    disable_unsafe_disclaimer: bool,
+}
+
+impl OctezClientBuilder {
+    pub fn new() -> Self {
+        OctezClientBuilder::default()
+    }
+    pub fn set_path(mut self, path: PathBuf) -> Self {
```

Comment:
Nit: add empty line in between `fn` definitions.
Note: This should be part of the formatter. Tracked: https://linear.app/tezos/issue/JSTZ-154/update-formatter-to-add-empty-line-in-between-fn-definitions-in-impl